alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt

```diff
@@ -499,6 +544,44 @@ class TransportIndexWorkflowAction @Inject constructor(
                     )
                     return
                 }
+
+                val updatedWorkflow = request.workflow.copy(id = indexResponse.id)
+                val executionId = CompositeWorkflowRunner.generateExecutionId(false, updatedWorkflow)
+
+                val (workflowMetadata, _) = WorkflowMetadataService.getOrCreateWorkflowMetadata(
+                    workflow = updatedWorkflow,
+                    skipIndex = false,
+                    executionId = executionId
+                )
+
+                val delegates = (updatedWorkflow.inputs[0] as CompositeInput).sequence.delegates.sortedBy { it.order }
+                val monitors = monitorCtx.workflowService!!.getMonitorsById(delegates.map { it.monitorId }, delegates.size)
+
+                for (monitor in monitors) {
+                    val (monitorMetadata, created) = MonitorMetadataService.getOrCreateMetadata(
+                        monitor = monitor,
+                        createWithRunContext = true,
+                        workflowMetadataId = workflowMetadata.id
+                    )
+
+                    if (created == false && monitor.monitorType == Monitor.MonitorType.DOC_LEVEL_MONITOR) {
+                        val updatedMetadata = MonitorMetadataService.recreateRunContext(monitorMetadata, monitor)
+                        client.suspendUntil<Client, BulkByScrollResponse> {
+                            DeleteByQueryRequestBuilder(client, DeleteByQueryAction.INSTANCE)
+                                .source(monitor.dataSources.queryIndex)
+                                .filter(QueryBuilders.matchQuery("monitor_id", monitor.id))
+                                .execute(it)
+                        }
+                        docLevelMonitorQueries.indexDocLevelMonitorQueries(
+                            monitor,
+                            monitor.id,
+                            updatedMetadata,
+                            request.refreshPolicy,
+                            indexTimeout
+                        )
```

Comment:
why do we need to reindex the monitor queries?

Reply:
fixed it.